### ISSUE TITLE
collect reviews

### ISSUE THREAD

**Issue by ineagu** [Codeinwp/jft-assistant#2]
the plugin will also collect reviews for themes, after a theme is active for >7 days, if exist on jft, ask users how they would rate it.

**Comment by selul** [comment]
@contactashish13 let's tackle this one. 

Some thoughts here: 

1) The review will be an admin notice, triggered in JS. We can do a smart check and check if there are reviews already shown in the admin dashboard, either hide others and show ours or put ours first. You can target the notices based on their classes. 
2) The notice will have a message and two actions buttons, one for start rating and one for don't show this again.
3) When they click start rating they will get some kind of stars rating and a textarea to leave a comment.
4) When they send the review, this will be sent as a comment to the post id for that theme that they rate. You can use the core wp rest comment endpoint. 

Let  me know if you have any questions.

**Comment by ineagu** [comment]
@selul just to encourage users more to connect with the site, I would redirect people on a page on jft to leave the review, there we have more freedom to ask for details.

**Comment by selul** [comment]
yep, much better if i think so. 

@contactashish13 we can page on JFT where we pass the theme and generate dynamically the form.

**Comment by contactashish13** [comment]
@selul On jft, can I assume

1. a pre-created page with a pre-created form will be displayed
2. some form element (maybe a drop down) will be populated with the name of the theme coming from the jft assistant
3. on the basis of that form element, other elements can be hidden or shown.

**Comment by selul** [comment]
@contactashish13 yep, the URL that the user will be redirected will contain the theme_id from JFT.

On the landing page, you can create a simple form(can be a shortcode)  with a textarea and a 1to10 score.  The review will be a comment to that particular post represented by the theme.

**Comment by contactashish13** [comment]
@selul I was thinking of using GravityForms because that is already there. So if we create a form using GF and tag one if it's elements as a conditional element (e.g. the theme slug), it can control the display/hiding of other dependent elements. Would that work for you?

**Comment by selul** [comment]
@contactashish13 i'm not sure why do we need to hide form elements based on the slug ?

**Comment by contactashish13** [comment]
So that you can display different fields/questions depending on the theme. Even if that is not required, gravity forms can be used.

**Comment by selul** [comment]
yep, we can use gravity forms, but the same form will be used across all themes for now. only the after submission behaviour is different, i.e add the form entry as a comment to that specific theme.

**Comment by contactashish13** [comment]
@selul what will be the elements of the form?

**Comment by selul** [comment]
just a textarea and a 1to10 rating option.

**Comment by contactashish13** [comment]
1. will there be a field to show the theme name? will it be a drop down or plain HTML?
2. will the rating be a set of radio buttons as GF doesnt support a rating field?

**Comment by selul** [comment]
1) we can show like a message, form title where we should use the theme name -> Something like `Please review <theme_name>` ( plain html ) 
2) Yep, a set of radios works fine.

**Comment by contactashish13** [comment]
@selul have addressed the PR comments.
